### PR TITLE
feat(uat):  create thing for mqtt client

### PR DESCRIPTION
**Pull request by esmirnov-klika-tech**
**Description of changes:**
- added control engine to tests
- added step to configure thing certificate on agent 

**Why is this change necessary:**
- to use in feature testing

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: uat/pom.xml [hunk 1]
```diff
@@ -62,5 +62,8 @@
         <spotbugs.plugin.version>4.0.0</spotbugs.plugin.version>
         <antrun.plugin.version>3.0.0</antrun.plugin.version>
         <surefire.plugin.version>3.0.0</surefire.plugin.version>
+        <junit.version>5.9.2</junit.version>
```
Comment (bgklika) [inline uat/pom.xml:65]: In the latest uat/pom.xml  junit.version and mockito.version are already defined. Did you rebase your branch on top of latest uat-dev ?

File: uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java [hunk 2]
```diff
@@ -27,22 +35,55 @@ public class MqttControlSteps {
 
     private static final String DEFAULT_CLIENT_DEVICE_POLICY_CONFIG = "/configs/iot/basic_client_device_policy.yaml";
 
+    private static final int DEFAULT_CONTROL_GRPC_PORT = 47_619;
```
Comment (bgklika) [inline uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java:38]: It is possible to use autoselected port and pass it to client's config ? 
At least in future.

File: uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java [hunk 3]
```diff
@@ -74,10 +116,18 @@ public void createClientDevice(String clientDeviceId) {
...
+     * Pass IoT Thing certificate for connection into specified client component.
+     *
+     * @param clientDeviceId string user defined client device id
+     * @param componentId    componentId of MQTT client
+     * @param brokerId       broker id
+     */
     @And("I connect device {string} on {word} to {string}")
     public void connect(String clientDeviceId, String componentId, String brokerId) {
-        //@TODO Implement method
+        final MqttConnectRequest request = getMqttConnectRequest(clientDeviceId, componentId, brokerId);
```
Comment (bgklika) [inline uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java:128]: It is possible to have more than one connection at the same time ?
For different agents.
For the same agent.

File: uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java [hunk 4]
```diff
@@ -116,18 +166,75 @@ public void discoverCoreDeviceBroker(String string) {
         //@TODO Implement method
     }
```
Comment (bgklika) [inline uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java:167]: How connection we checked will be identified ?

**Review by bgklika — APPROVED** [review]

File: uat/pom.xml (hunk 1 above)
Comment (bgklika) [reply uat/pom.xml:65]: Strictly speaking junit.version already defined in line 48
nd mockito.version on line 49

**Review by bgklika — APPROVED** [review]

File: uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java (hunk 3 above)
Comment (esmirnov-klika-tech) [reply uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java:128]: yes for both cases

File: uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java (hunk 4 above)
Comment (esmirnov-klika-tech) [reply uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java:167]: Currently we agreed that connectionId == iot thing id

File: uat/pom.xml (hunk 1 above)
Comment (esmirnov-klika-tech) [reply uat/pom.xml:65]: Will be fixed in the next version )

File: uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java (hunk 2 above)
Comment (esmirnov-klika-tech) [reply uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java:38]: It is possible

**Pull request closed without merging**